lms/static/scripts/frontend_apps/components/dashboard/DashboardActivityFilters.tsx

```diff
@@ -155,12 +151,12 @@ export default function DashboardActivityFilters({
 
   return (
     <div className="flex gap-2 flex-wrap">
-      <MultiSelect
+      <Select
         disabled={coursesResult.isLoadingFirstPage}
-        value={selectedCourseIds}
-        onChange={newCourseIds =>
+        value={selectedCourseIds[0]}
```

Comment:
This subtly relies on `array[index]` returning undefined if the index is out of bounds. The type however is still inferred as `string`, which is misleading because in the `onChange` callback the argument type is inferred as `string` but is actually `string|undefined`.

One way to resolve this is to use a simple helper, eg:

```
const firstItem = <T,>(array: T[]) => array[0] as T | undefined;
                                                                 
return (
  <div className="flex gap-2 flex-wrap">
    <Select
      disabled={coursesResult.isLoadingFirstPage}
      value={firstItem(selectedCourseIds)}
```

Reply:
Yes, type inference is not correct, it should be inferring `string | undefined` both here and in `onChange`.

However, I'm already implementing the logic taking that into consideration. See `onCHange`, where we do `courses.onChange(newCourseId ? [newCourseId] : [])`.

In order to get proper type inference, I think we need to change [an option](https://www.typescriptlang.org/tsconfig/#noUncheckedIndexedAccess) in tsconfig, but we tried once and it had other side effects, so we deferred it.

Reply:
> In order to get proper type inference, I think we need to change [an option](https://www.typescriptlang.org/tsconfig/#noUncheckedIndexedAccess) in tsconfig, but we tried once and it had other side effects, so we deferred it.

That's a way of making this work globally, but using a helper as in my code snippet above is a way of achieving this locally.

Reply:
True. I think it's worth adding your suggestion.